### PR TITLE
Change keyboard navigation

### PR DESCRIPTION
**Pull request by siemiatj**
Related to #2436

### CODE REVIEW

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh-webui-frontend/metasfresh-webui-frontend/pullRequest?prid=4578381&bid=15381754)

File: src/components/widget/RawWidget.js [hunk 1]
```diff
@@ -401,7 +406,7 @@ export class RawWidget extends Component {
       disabled: readonly,
       onFocus: this.handleFocus,
       tabIndex: tabIndex,
-      onChange: e => handleChange && handleChange(widgetField, e.target.value),
+      onChange: e => handleChange && handleChange(widgetField, e.target.value, cachedValue || widgetValue),
```
Comment (metas-ts) [inline src/components/widget/RawWidget.js]: Codacy Issue found: [Replace `·handleChange·&&` with `⏎········handleChange·&&⏎·······` (prettier/prettier)](https://app.codacy.com/app/metasfresh/metasfresh-webui-frontend/pullRequest?prid=4578381)

File: src/components/table/TableItem.js [hunk 2]
```diff
@@ -365,8 +415,13 @@ class TableItem extends PureComponent {
...
                 ref={c => {
-                  if (c && isSelected && focusOnFieldName === property) {
-                    this.autofocusCell = c;
+                  if (c && isSelected) {
+                    if (focusOnFieldName === property) {
+                      this.autofocusCell = c;
+                    }
+                    if (activeCellName === property) {
+                      this.selectedCell = c;
+                    }
```
Comment (metas-ts) [inline src/components/table/TableItem.js]: Codacy Issue found: [Insert `;` (prettier/prettier)](https://app.codacy.com/app/metasfresh/metasfresh-webui-frontend/pullRequest?prid=4578381)

File: src/components/table/TableItem.js [hunk 3]
```diff
@@ -107,39 +140,51 @@ class TableItem extends PureComponent {
           changeListenOnTrue();
         }
         break;
+      default: {
+        const inp = String.fromCharCode(e.keyCode);
+        if (/[a-zA-Z0-9]/.test(inp)) {
+
```
Comment (metas-ts) [inline src/components/table/TableItem.js]: Codacy Issue found: [Replace `⏎····const·extendLongText·=·3;⏎` with `····const·extendLongText·=·3;` (prettier/prettier)](https://app.codacy.com/app/metasfresh/metasfresh-webui-frontend/pullRequest?prid=4578381)

File: src/components/table/TableCell.js [hunk 4]
```diff
@@ -394,9 +401,11 @@ TableCell.propTypes = {
...
-export default connect(state => ({
+const mapStateToProps = state => ({
   modalVisible: state.windowHandler.modal.visible,
   isGerman: state.appHandler.me.language
     ? state.appHandler.me.language.key.includes('de')
     : false,
-}))(TableCell);
+});
+
+export default connect(mapStateToProps, null, null, { forwardRef: true })(TableCell);
```
Comment (metas-ts) [inline src/components/table/TableCell.js]: Codacy Issue found: [Replace `mapStateToProps,·null,·null,·{·forwardRef:·true·}` with `⏎··mapStateToProps,⏎··null,⏎··null,⏎··{·forwardRef:·true·}⏎` (prettier/prettier)](https://app.codacy.com/app/metasfresh/metasfresh-webui-frontend/pullRequest?prid=4578381)

File: src/components/widget/MasterWidget.js [hunk 5]
```diff
@@ -31,23 +31,16 @@ class MasterWidget extends Component {
     data: '',
   };
 
-  /**
-   * @method componentDidMount
-   * @summary ToDo: Describe the method.
-   */
   componentDidMount() {
-    const { data, widgetData } = this.props;
+    const { data, widgetData, clearValue } = this.props;
```
Comment (metas-ts) [inline src/components/widget/MasterWidget.js]: Codacy Issue found: ['clearValue' is missing in props validation (react/prop-types)](https://app.codacy.com/app/metasfresh/metasfresh-webui-frontend/pullRequest?prid=4578381)

File: src/components/widget/MasterWidget.js [hunk 6]
```diff
@@ -167,7 +156,7 @@ class MasterWidget extends Component {
    * @param {*} property
    * @param {*} val
    */
-  handleChange = (property, val) => {
+  handleChange = (property, val, cachedVal) => {
```
Comment (metas-ts) [inline src/components/widget/MasterWidget.js]: Codacy Issue found: ['cachedVal' is defined but never used. (no-unused-vars)](https://app.codacy.com/app/metasfresh/metasfresh-webui-frontend/pullRequest?prid=4578381)

File: src/components/table/TableCell.js [hunk 7]
```diff
@@ -215,6 +219,10 @@ class TableCell extends PureComponent {
     }
   };
 
+  clearValue = reset => {
+    this.clearWidgetValue = reset == null ? true : false;
+  }
```
Comment (metas-ts) [inline src/components/table/TableCell.js]: Codacy Issue found: [Replace `····` with `········` (prettier/prettier)](https://app.codacy.com/app/metasfresh/metasfresh-webui-frontend/pullRequest?prid=4578381)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh-webui-frontend/metasfresh-webui-frontend/pullRequest?prid=4578381&bid=15381754)

File: src/components/widget/MasterWidget.js (hunk 5 above)
Comment (metas-ts) [inline src/components/widget/MasterWidget.js]: Codacy Issue found: ['clearValue' is missing in props validation (react/prop-types)](https://app.codacy.com/app/metasfresh/metasfresh-webui-frontend/pullRequest?prid=4578381)

Comment (siemiatj) [reply src/components/widget/MasterWidget.js]: I want to have a green build ASAP, so I'll add this prop in the follow-up

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh-webui-frontend/metasfresh-webui-frontend/pullRequest?prid=4578381&bid=15381754)

**Pull request merged**